lib/utils/RequestHandler.js

```diff
@@ -16,7 +16,10 @@ class RequestHandler {
         this.gridApiStartPath = '/grid/api'
         this.eventHandler = eventHandler
         this.logger = logger
-        this.defaultOptions = options
+        this.options = options
```

Comment:
`this.options` are shared between `webdriverio` instance and `requestHandler` so I suggest to do not touch `RequestHandler` at all. Lets set connection retry options in `webdriverio` instance.